src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java

```diff
@@ -19,449 +19,11 @@
 
 package com.puppycrawl.tools.checkstyle;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+/**
+ * @deprecated Checkstyle will not support BaseCheckTestSupport anymore. Use
+ *             {@link BaseTestSupport} instead.
+ */
+@Deprecated
+public class BaseCheckTestSupport extends BaseTestSupport {
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.LineNumberReader;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.text.MessageFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.ResourceBundle;
-import java.util.stream.Collectors;
-
-import com.google.common.collect.MapDifference;
-import com.google.common.collect.MapDifference.ValueDifference;
-import com.google.common.collect.Maps;
-import com.puppycrawl.tools.checkstyle.api.Configuration;
-import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
-
-public class BaseCheckTestSupport {
```

Comment:
@subkrish to keep history of this file split this into 2 commits.
1st commit renames old class to new class.
2nd commit adds old class back, but just empty.